### ISSUE TITLE
"Edit Node" brings Internal JavaScript Error

### ISSUE THREAD

**Issue by joekroese** [felixhayashi/TW5-TiddlyMap#347]
I use TW5-TiddlyMap as part of my [TiddlyRoam](https://github.com/joekroese/tiddlyroam) project. (In fact, it pretty much is one half of the project!)

Unfortunately, we are having an issue with 'Edit Node' which brings about a javascript error joekroese/tiddlyroam#11.

<img width="873" alt="Screenshot 2020-05-09 at 10 09 11" src="https://user-images.githubusercontent.com/7649772/81468382-8139ba00-91df-11ea-95d0-584a62199547.png">

Clicking "Edit Node" while looking at the Chrome developer window shows the more detailed error message:

```
_$:/plugins/felixhayashi/vis/vis.js:65_
Invalid type received for "physics". Expected: boolean. Received [object] "[object Object]"

Problem value found at:
options = {
  nodes: {
    physics
  }
}


_$:/plugins/felixhayashi/vis/vis.js:86_
Errors have been found in the supplied options object.


_tiddlyroam_test.html:12808_
Uncaught TypeError: Cannot read property 'forEach' of null

$tw.utils.error | @ | tiddlyroam_test.html:12808
window.onerror | @ | tiddlyroam_test.html:12835
window.onerror | @ | $:/plugins/felixhayashi/vis/vis.js:30
```
Are you able to see what might be the problem?

**Comment by felixhayashi** [comment]
Hi @joekroese, 

thanks for reporting, I will have a look at it. When I installed the latest version of TiddlyMap in TiddlyWiki 5.1.22 I did not experience this issue (maybe could you confirm this?). Therefore I assume the error is caused by something in combination with TiddlyRoam or the way it is setup :thinking: I will try to reproduce.

P.S. Congrats on setting up tiddlyroam, it seems to have gained quite some popularity in such a short time :slightly_smiling_face:  :+1:

**Comment by joekroese** [comment]
Thanks for the kind words @felixhayashi. It's nice to be able to repackage the good work of others into something else that people find useful.

I can confirm that TiddlyMap plays nicely with TiddlyWiki 5.1.22. It's almost definitely something I introduced with TiddlyRoam. Unfortunately, I just can't tell what!

It's part of a bigger issue of mine that is I haven't learnt the best practices for developing for TiddlyWiki yet. I'm taking inspiration from your work with TiddlyMap for improving that going forwards. Baby steps!

**Comment by gtothill** [comment]
Hi all,

Just to jump on this thread with a report. I notice that the screencap @joekroese posted seems to be different to the one I'm getting. I also see an error as described in his tiddlyroam post vis:

![image](https://user-images.githubusercontent.com/52030855/82319776-1734c800-99ca-11ea-8843-eda7aebb93c8.png)

I seem to be able to reproduce this in an empty tiddlywiki with the following components:

TW5 Core 5.1.22
hotzone: 0.0.3
tiddlymap: 0.17.4+9899
topstoryview: 0.2.0
vis: as version on tiddlymap site.

This is some debug:

`Invalid type received for "physics". Expected: boolean. Received [object] "[object Object]"Problem value found at: options = {  nodes: {    physics  }}`

Strangely, in wiki's created last week I do not see this behaviour - even with apparently identical plugins.

Also - this only seems to happen in live view nodes, not others.

Hope this helps,

Regards,

Geoff.

**Comment by felixhayashi** [comment]
Hi guys, I was able to track down the moment this happens in TiddlyMap and will fix it soon – but I am still puzzled why this does only occurs with the TiddlyRoam setup :man_shrugging: 

Anyhow, will post here once the new version is released.

**Comment by joekroese** [comment]
Nice work @felixhayashi - thanks for looking into this.

Yep, it is definitely strange that it happens with TiddlyRoam. Does @gtothill's report make things more or less clear as to why it's happening? Either way, looking forward to the new version when you get round to that 🙌

**Comment by felixhayashi** [comment]
@joekroese Your comment made me re-read @gtothill 's comment again and I had somehow overlooked that he is not using TiddlyRoam as dependency and this error still occurs. I was now also able to reproduce this in tiddlymap.org – it happens in situations where the **view's** physics configuration were changed. This is the case per default in the "live view", where I did not expect people to hit the "edit node" button. In TiddlyRoam switching the view to something else (e.g. the default view) "edit node" does not cause an error anymore.

So, again, thanks everybody for contributing to this issue, I will fix it soon this week.

**Comment by gtothill** [comment]
Excellent @felixhayashi. Sorry I didn't make it clearer that I wasn't using TiddlyRoam. I had that in my head as I wrote the post but for some reason it never made the keyboard :-)

**Comment by felixhayashi** [comment]
fixed with v0.17.5

**Comment by gtothill** [comment]
@felixhayashi - thank you!
